src/main/java/tudee/ui/Ui.java

```diff
@@ -0,0 +1,144 @@
+package tudee.ui;
+
+import java.util.List;
+import java.util.Scanner;
+
+import tudee.task.Task;
+import tudee.task.TaskList;
+
+/**
+ * The Ui class handles the interaction with the user.
+ * It manages input and output, displaying messages to the user,
+ */
+public class Ui {
+    private static final String STRAIGHTLINE = "____________________________________________________________ \n";
+    private final Scanner sc;
+
+    /**
+     * Constructs a new Ui object and initializes the scanner
+     * for reading user input.
+     */
+    public Ui() {
+        sc = new Scanner(System.in);
+    }
+
+    /**
+     * Prints a straight line to the console to separate sections of output.
+     */
+    public void showLine() {
+        System.out.println(STRAIGHTLINE);
+    }
+
+    /**
+     * Displays a welcome message to the user.
+     */
+    public void showWelcome() {
+        showLine();
+        System.out.println("Hello! I'm Tudee, your chatbot bestie! \nHow can I help you today? :) \n");
+        showLine();
+    }
+
+    /**
+     * Displays a goodbye message to the user.
+     */
+    public String showBye() {
+        return "Bye. Hope to see you again soon! Don't forget about me :( \n";
+    }
+
+    /**
+     * Displays the current list of tasks to the user.
+     *
+     * @param tasks The list of tasks to be displayed.
+     */
+    public String showTaskList(List<Task> tasks) {
+        String output = "";
+        for (int i = 0; i < tasks.size(); i++) {
+            output += i + 1 + ". " + tasks.get(i) + "\n";
+        }
+        return output;
+    }
+
+    /**
+     * Displays a message indicating that a task has been marked as done.
+     *
+     * @param task The task that has been marked as done.
+     */
+    public String showMark(Task task) {
+        return "Nice! I've marked this task as done: \n" + task;
+    }
+
+    /**
+     * Displays a message indicating that a task has been marked as not done.
+     *
+     * @param task The task that has been marked as not done.
+     */
+    public String showUnMark(Task task) {
```

Comment:
Method name could be potentially changed to `showUnmarkedTask` to better describe the method.